psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,214 @@
+        for component in salc_list[salc_index]:
+            geom.add(0, component.atom, component.xyz,
+                     disp_steps * step_size * component.coef / math.sqrt(mol.mass(component.atom)))
+    # Now let's get rid of that initial ", ".
+    geom.name = name[2:]
+
+
+def geom_generator(mol, freq_irrep_only, mode):
+    """
+    Generate geometries for a finite difference calculation and print information about the geometries generated.
+    You probably want to use the fd_geoms_1_0, fd_geoms_freq_2_0, or fd_geoms_freq_2_1 convenience functions. 
+
+    mol: The molecule to generate displaced geometries for
+    freq_irrep_only: An int specifying the Cotton ordered irrep to take frequencies for.
+    mode: A string that specifies the quantities determined at and from the displacements. See msg_dist.
+    """
+
+    print_lvl = core.get_option("FINDIF", "PRINT")
+    num_pts = core.get_option("FINDIF", "POINTS")
+    disp_size = core.get_option("FINDIF", "DISP_SIZE")
+
+    msg_dict = {
+        "1_0":
+        "energies to determine gradients",
+        "2_1":
+        "gradients to determine vibrational frequencies and \n"
+        "  normal modes.  Resulting frequencies are only valid at stationary points.",
+        "2_0":
+        "gradients to determine vibrational frequencies and \n"
+        "  normal modes.  Resulting frequencies are only valid at stationary points."
+    }
+
+    try:
+        print_msg = msg_dict[mode]
+    except KeyError:
+        raise ValidationError("FINDIF: Mode {} not recognized.".format(mode))
+
+    if print_lvl:
+        core.print_out("\n-------------------------------------------------------------\n\n"
+                       "  Using finite-differences of {:s}\n"
+                       "\tGenerating geometries for use with {:d}-point formula.\n"
+                       "\tDisplacement size will be {:6.2e}.\n".format(print_msg, num_pts, disp_size))
+
+    # TODO: Replace with a generator from a stencil to a set of points.
+    pts_dict = {
+        3: {
+            "sym": ((-1, ), (1, )),
+            "asym": ((-1, ), ),
+            "off": ((1, 1), (-1, -1))
+        },
+        5: {
+            "sym": ((-2, ), (-1, ), (1, ), (2, )),
+            "asym": ((-2, ), (-1, )),
+            "off": ((-1, -2), (-2, -1), (-1, -1), (1, -1), (-1, 1), (1, 1), (2, 1), (1, 2))
+        }
+    }
+
+    # We won't need this variable for a while, but let's validate num_pts early.
+    try:
+        disps = pts_dict[num_pts]
+    except KeyError:
+        raise ValidationError("FINDIF: Invalid number of points!")
+
+    # Get settings for CdSalcList, then get it.
+    method_allowed_irreps = 0x1 if mode == "1_0" else 0xFF
+    t_project = not core.get_global_option("EXTERN") and not core.get_global_option("PERTURB_H")
+    # core.get_local_option returns an int, but CdSalcList expect a bool... Re-cast as bool.
+    r_project = t_project and bool(core.get_local_option("FINDIF", "FD_PROJECT"))
```

Comment:
I think you want `get_option('FINDINF', 'FD_PROJECT')` here. Several people are desirous of killing off the Options object rather than explaining why, but why will be explained upon request.